docs/decisions/0009-pylint-assertions.rst

```diff
@@ -0,0 +1,50 @@
+Pylint Assertions Standardization
+---------------------------------
+
+Status
+======
+
+Proposed
+
+Context
+=======
+
+The edx-platform test suite includes a massive number of individual assertions, written over time for different Python versions and test runners.  A nontrivial number of these currently trigger deprecation warnings, as they are slated for replacement or removal in future Python versions.  A very high percentage of them use assertion methods from Python's ``unittest`` module, which were inspired by Java's jUnit library long ago and have notable drawbacks vs. the `assert statement rewriting`_ built into our current test runner, pytest.  In particular:
+
+* ``assert`` statements are typically shorter and easier to read.  For example:
+
+  * ``self.assertNotIn("session_key", self.client.session)`` (unittest) becomes
+
+  * ``assert "session_key" not in self.client.session`` (pytest)
+
+* The output of failed pytest-rewritten assertions is often more helpful than their ``unittest`` equivalents.
```

Comment:
This is highly subjective. As far as I am concerned, pytest is too verbose, to the point that it makes it very difficult to detect relevant information whenever a test fails.